lib/sles4sap/sap_host_agent.pm

```diff
@@ -0,0 +1,102 @@
+# SUSE's openQA tests
+#
+# Copyright 2017-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Functions for SAP tests
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_host_agent;
+use strict;
+use warnings;
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+
+our @EXPORT = qw(
+  saphostctrl_list_databases
+  parse_instance_name
+);
+
+my $saphostctrl = '/usr/sap/hostctrl/exe/saphostctrl';
+
+=head1 SYNOPSIS
+
+Package with functions related to interaction with SAP Host Agent (Command B<saphostctrl>). Those can be used for collecting
+data about instances and performing various operations. Keep in mind that command needs to be executed using either
+B<root> or B<sidadm>.
+
+=cut
+
+=head2 saphostctrl_list_databases
+
+    saphostctrl_list_databases([as_root=>1]);
+
+Uses C<saphostctrl> to get list of all databases residing on host. Returns parsed output as B<ARRAYREF>.
+Data for each DB is contained in a B<HASH>
+Example:
+$VAR1 = {
+          'Hostname' => 'qesdhdb01l029',
+          'Release' => '2.00.075.00.1702888292',
+          'Vendor' => 'HDB', # HDB = SAP hana database (not SID)
+          'Type' => 'hdb', # type of hana DB - hdb, mdc (multitenant), systemdb
+          'Instance name' => 'PRD00'
+        };
+$VAR2 = {
+          'Hostname' => 'qesdhdb01l029',
+          'Type' => 'hdb',
+          'Release' => '2.00.075.00.1702888292',
+          'Instance name' => 'QAS01',
+          'Vendor' => 'HDB'
+        };
+
+=over
+
+=item * B<as_root>: Execute command using sudo. Default: false
+
+=back
+=cut
+
+sub saphostctrl_list_databases {
+    my (%args) = @_;
+    die("Missing saphostctrl binary: $saphostctrl")
+      if script_run("[ -f $saphostctrl ]");
```

Comment:
I like you did this as the error message is better for reviewers, but you didn't need `die if script_run "[ -f $saphostctrl ]"`, you could've used directly `assert_script_run "[ -f $saphostctrl ]"`

Reply:
Oh thanks... that's neat.